Src/VimCore/EditorUtil.fs

```diff
@@ -21,18 +21,60 @@ module internal EditorCoreUtil =
     let IsEndPoint (point: SnapshotPoint) = 
         point.Position = point.Snapshot.Length
 
+    /// TODO: Should this handle surrogate pairs and linebreaks?
     let AddOneOrCurrent (point: SnapshotPoint) =
         if IsEndPoint point then
             point
         else
             point.Add(1)
 
+    /// TODO: Should this handle surrogate pairs and linebreaks?
     let SubtractOneOrCurrent (point: SnapshotPoint) = 
         if point.Position = 0 then
             point
         else
             point.Subtract(1)
 
+    /// Get the span of the character which is pointed to by the point.  Normally this is a 
+    /// trivial operation.  The only difficulty is if the point exists at the end of a line
+    /// (in which case the span covers the linebreak) or if the character is part of a
+    /// surrogate pair.
+    let GetCharacterSpan (line: ITextSnapshotLine) (point: SnapshotPoint) = 
+
+        // We require that the point belong to the line.
+        Contract.Requires(point.Snapshot = line.Snapshot)
+        Contract.Requires(point.Position >= line.Start.Position)
+        Contract.Requires(point.Position <= line.EndIncludingLineBreak.Position)

```

Comment:
Can't we change the input here to a `SnapshotColumn` and avoid all of these asserts?